src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/UnionLayoutComputer.java

```diff
@@ -77,15 +77,19 @@ MemoryLayout fieldLayout(Cursor c) {
     long fieldSize(Cursor c) {
         if (c.type().kind() == TypeKind.IncompleteArray) {
             return 0;
+        } else if (c.isBitField()) {
+            return c.getBitFieldWidth();
+        } else {
+            return c.type().size() * 8;
         }
-        return c.type().size() * 8;
     }
 
     @Override
     MemoryLayout finishLayout() {
-        // size mismatch indicates anonymous bitfield used for padding
+        // size mismatch indicates use of bitfields in union
         long expectedSize = type.size() * 8;
         if (actualSize < expectedSize) {
+            // emit an extra padding of expected size to make sure union layout size is computed correctly
             addFieldLayout(MemoryLayout.ofPaddingBits(expectedSize));
```

Comment:
Maybe an assert could be added here to check that `actualSize` is exactly `expectedSize` at the end. I think we have a problem otherwise.

Reply:
actual size can be less if there are bitfields - in which case we need to emit a padding word to make sure the union layout has right size.

Reply:
Yes, I'm wondering if there can be cases where the actualSize is larger than the expected size, which isn't checked here currently.

Reply:
I'll add an assertion